addons/web_editor/static/src/js/editor/odoo-editor/src/utils/sanitize.js

```diff
@@ -21,53 +21,48 @@ import {
 } from './utils.js';
 
 const NOT_A_NUMBER = /[^\d]/g;
+
+function _hasPseudoElementContent (node, pseudoSelector) {
+    const value = getComputedStyle(node, pseudoSelector).getPropertyValue('content');
+    return value && value !== 'none';
+}
+
 export function areSimilarElements(node, node2) {
-    if (
-        !node ||
-        !node2 ||
-        node.nodeType !== Node.ELEMENT_NODE ||
-        node2.nodeType !== Node.ELEMENT_NODE
-    ) {
-        return false;
-    }
-    if (node.tagName !== node2.tagName) {
-        return false;
+    if (![node, node2].every(n => n?.nodeType === Node.ELEMENT_NODE)) {
+        return false; // The nodes don't both exist or aren't both elements.
     }
-    for (const att of node.attributes) {
-        const att2 = node2.attributes[att.name];
-        if ((att2 && att2.value) !== att.value) {
-            return false;
-        }
-    }
-    for (const att of node2.attributes) {
-        const att2 = node.attributes[att.name];
-        if ((att2 && att2.value) !== att.value) {
-            return false;
-        }
+    if (node.nodeName !== node2.nodeName) {
+        return false; // The nodes aren't the same type of element.
     }
-    function isNotNoneValue(value) {
-        return value && value !== 'none';
+    const nodeName = node.nodeName;
+
+    // Compare the node's attributes.
+    const [attrs, attrs2] = [node, node2].map(n => [...n.attributes].reduce((object, attribute) => {
+        object[attribute.name] = attribute.value;
+        return object; // Make two objects with the attributes of each node as key/value pairs.
+    }, {}));
+    if (Object.entries(attrs).sort().toString() !== Object.entries(attrs2).sort().toString()) {
+        return false; // The nodes don't have the same attributes.
```

Comment:
> That lead me to this which circumvents the conversion to object and thus the reduce

Really well played :+1:

> it will fail to return `false` for these two elements: `<div a="b" c="d" e="f">` `<div a="b,c,d" e="f">`

Wow eye of the tiger :ok_hand: Wouldn't that be solved by sorting first, then replacing the condition with `attrs.length !== attrs2.length || !attrs.every((attr, i) => attr[0] === attrs2[i][0] && attr[1] === attrs2[i][1])` ? If we wrap the whole thing in a `if (node.attributes.length === node2.attributes.length)` we can remove the `||` .
That said I see your points about complexity and it's maybe indeed overall not worth it.

This would leave us with something like this:
```js
let [attrs, attrs2] = [node.attributes, node2.attributes];
if (attrs.length === attrs2.length) {
    [attrs, attrs2] = [attrs, attrs2].map(a => [...a].map(b => [b.name, b.value]).sort());
    if (!attrs.every((a, i) => a[0] === attrs2[i][0] && a[1] === attrs2[i][1])) {
        return false;
    }
}
```

I think at the end of the day the clearest is this suggestion of yours:
```js
for (const attr of node.attributes) {
    if (attr.value !== node2.attributes[attr.name]?.value) {
        return false;
    }
}
for (const attr2 of node2.attributes) {
    if (attr2.value !== node.attributes[attr2.name]?.value) {
        return false;
    }
}
```

I'm just annoyed with the duplication but I think that's on me, I have to learn to live with it :-P

Reply:
> I'm just annoyed with the duplication but I think that's on me, I have to learn to live with it :-P

If the duplication is an issue then I'm surprised you didn't choose the last version :p

```js
for (const name of new Set([
    ...node.getAttributeNames(),
    ...node2.getAttributeNames(),
])) {
    if (node.getAttribute(name) !== node2.getAttribute(name)) {
        return false;
    }
}
```

Reply:
Thing is, the duplication is _not_ an issue. Just something that irks me :-P At the end of the day, readability is more important, and I find that last version less easy to understand at a glance. Although on second thought I think that's just because I'm more used to see the other form and this is actually find. So whatever you prefer I guess :-P

Reply:
```js
const attrNames = new Set([...node.getAttributeNames(), ...node2.getAttributeNames()]);
if ([...attrNames].some(name => node.getAttribute(name) !== node2.getAttribute(name)) {
    return false;
}
```
is also possible ^^'

Reply:
Honestly, I think those last two propositions are both less readable than the other two contenders... ^^'
Choose whichever you prefer between:
```js
for (const attr of node.attributes) {
    if (attr.value !== node2.attributes[attr.name]?.value) {
        return false;
    }
}
for (const attr2 of node2.attributes) {
    if (attr2.value !== node.attributes[attr2.name]?.value) {
        return false;
    }
}
```
and
```js
for (const name of new Set([
    ...node.getAttributeNames(),
    ...node2.getAttributeNames(),
])) {
    if (node.getAttribute(name) !== node2.getAttribute(name)) {
        return false;
    }
}
```